### PR TITLE
fix: handle bitvectors in type inference Join/Meet

### LINKED ISSUE #1277: Crash in type inference for bitvectors
Dafny crashes (in the `JoinX` method in the resolver) given the following program:

``` dafny
function RangeMask(from: nat, to: nat): bv6

lemma CalcEqBits(i: nat, j: nat)
{
  calc == {
    RangeMask(i, j);
    -1 << (6 - j);
  }
}
```

### PR DESCRIPTION
**Pull request by RustanLeino**
Fixes #1277

### CODE REVIEW

File: Source/Dafny/DafnyAst.cs [hunk 1]
```diff
@@ -1900,6 +1900,10 @@ public enum AutoInitInfo { MaybeEmpty, Nonempty, CompilableValue }
       } else if (a.IsNumericBased()) {
         // Note, for meet, we choose not to step down to IntVarietiesSupertype or RealVarietiesSupertype
         return a.Equals(b) ? a : null;
+      } else if (a.IsBitVectorType) {
+        var aa = a.AsBitVectorType;
```
Comment (keyboardDrummer) [inline Source/Dafny/DafnyAst.cs:1904]: Can you not do something like `if (a.IsBitVectorType) return Equals(a,b) ? a : null`

File: Source/Dafny/DafnyAst.cs [hunk 2]
```diff
@@ -2126,6 +2130,10 @@ public enum AutoInitInfo { MaybeEmpty, Nonempty, CompilableValue }
         return a.IsNumericBased(NumericPersuasion.Real) ? a : null;
       } else if (a.IsNumericBased()) {
         return a.Equals(b) ? a : null;
+      } else if (a.IsBitVectorType) {
+        var aa = a.AsBitVectorType;
```
Comment (keyboardDrummer) [inline Source/Dafny/DafnyAst.cs:2134]: Can you not do something like `if (a.IsBitVectorType) return Equals(a,b) ? a : null`

File: Source/Dafny/DafnyAst.cs (hunk 1 above)
Comment (RustanLeino) [reply Source/Dafny/DafnyAst.cs]: Yes, that's better--and makes it look like the other cases.

File: Source/Dafny/DafnyAst.cs (hunk 2 above)
Comment (RustanLeino) [reply Source/Dafny/DafnyAst.cs]: Done

**Review by keyboardDrummer — APPROVED** [review]

**Pull request merged**